crates/block-explorers/src/lib.rs

```diff
@@ -70,9 +71,16 @@ impl std::fmt::Display for EtherscanApiVersion {
 impl TryFrom<String> for EtherscanApiVersion {
     type Error = EtherscanError;
 
-    #[inline]
     fn try_from(value: String) -> Result<Self, Self::Error> {
-        match value.as_str() {
+        Self::from_str(value.as_str())
+    }
+}
+
+impl FromStr for EtherscanApiVersion {
+    type Err = EtherscanError;
+
+    fn from_str(value: &str) -> Result<Self, Self::Err> {
+        match value {
```

Comment:
should we also check the uppercase variants "V1"?

Reply:
yep. makes sense to have both, will add

Reply:
added in https://github.com/foundry-rs/block-explorers/pull/85/commits/f77a91f6f0f2610266001ebc8f769b712769bb1e